sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -162,13 +164,13 @@ namespace Azure { namespace Core {
   /**
    * @brief An exception that gets thrown when @DateTime error occurs.
    */
-  class DateTimeException : public std::runtime_error {
-  public:
+  struct DateTimeException : public Azure::Core::RequestFailedException
```

Comment:
In readme, we say `Exceptions for reporting errors from service requests in a consistent fashion (`RequestFailedException`)`, if this is still true, DateTimeException does not neccessarily originate from a request, or a response. Do we have some more generic Azure::Core::Exception ?

Reply:
Well, it can be returned when calling an API rigth?  
I believe that's what we want with this RequestFailedException, that this is what all APIs would return. Then Services would create their exceptions on top of just this one only base exception.

The next more general exception would be runtime_error.

Also, if anyone tries to create a DateTime without calling any API, they would still get a DateTimeException directly.
So, it is still one exception that can be used directly,
However, if an API failed to create or use DateTime, it would catch the RequestFailedException to know that the exception is comming from Azure CORE, then we can use dynamic cast to figure it what exception it is